tests/ping.go

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package tests
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+
+	expect "github.com/google/goexpect"
+	"google.golang.org/grpc/codes"
+	"k8s.io/utils/net"
+
+	v1 "kubevirt.io/client-go/api/v1"
+)
+
+var (
+	promptPriv   = regexp.MustCompile("\\# ")
+	promptUnPriv = regexp.MustCompile("\\$ ")
+	shellSuccess = regexp.MustCompile("\n0")
+	shellFail    = regexp.MustCompile("\n[1-9].*")
+)
+
+// PingFromVMConsole performs a ping through the provided VMI console.
+// Optional arguments for the ping command may be provided, overwirting the default ones.
+// (default ping options: "-c 1, -w 5")
+// Note: The maximum overall command timeout is 10 seconds.
+func PingFromVMConsole(vmi *v1.VirtualMachineInstance, ipAddr string, args ...string) error {
+	const maxCommandTimeout = 10
+
+	pingString := "ping"
+	if net.IsIPv6String(ipAddr) {
+		pingString = "ping -6"
+	}
+
+	if len(args) == 0 {
+		args = []string{"-c 1", "-w 5"}
+	}
+	args = append([]string{pingString, ipAddr}, args...)
+	cmdCheck := strings.Join(args, " ") + "\n"
+
+	err := CheckForTextExpecter(vmi, []expect.Batcher{
```

Comment:
It will conflict with https://github.com/kubevirt/kubevirt/pull/3882/commits/a3d74ed0f3b4715c7232c7e6b928b05927a15220 which doesn't support `expect.Case`.
One of us will have the rebase conflict:) Maybe we can add a similar method to `CheckForTextExpecter` which is not using `ExpectBatchWithValidatedSend`.

Reply:
Why does it conflict? This one will not use (for now at least) the extra validation until it will be available and this one converted to use it.

Reply:
CheckForTextExpecter is using ExpectBatchWithSendValidation

Reply:
CheckForTextExpecter is using ExpectBatchWithSendValidation

Reply:
At this time, there seem to be nothing I can do with this.
It is also strange that you do not support all options, it may surprise someone who wants to use it.
When you say “no support”, you mean it will explode or not add the verify part to it?

Reply:
It will return an error it is not supported. And yes, currently there is nothing you can do about it. It was just a hint of what should be done in case my PR will be merged before yours:)